urwid/tests/test_vterm.py

```diff
@@ -60,6 +63,7 @@ def read(self, size: int) -> bytes:
 
     def write(self, data: bytes) -> None:
         os.write(self.writer, data)
+        sleep(0.025)
```

Comment:
Not fond of this. Never really fixes flaky tests — merely shifts flake probability.

Did you try `os.fdopen(self.writer).flush()` instead ?

Or better yet, disabling buffering completely with `unbuffered_pipe = os.fdopen(self.writer, 'wb', 0)

Reply:
We can try unbuffered pipe, but `stdout` will be still buffered

Reply:
`os.fdopen` fails in 10 from 10 runs with `Bad file descriptor`
`os.fsync` also disagree to deal with pipe

Reply:
That can be disabled too.

In a unit test though, why is there a dependency onto `sys.stdout` at all? Can't it allocate a private PTY ?

... Even that seems redundant, as `.read()` passes the output of `Terminal.render()` to asserts directly:

https://github.com/urwid/urwid/blob/bac20db87eaeb0b7a0026807d7e6dbe391b08580/urwid/tests/test_vterm.py#L118-L129

I don't see any need to use `sys.stdout` at all.

Reply:
Render spawn process, which execute `pty.fork()`.
Data read from fork stdout and rendered.